integration-libs/cdc/root/service/cdc-js.service.ts

```diff
@@ -170,81 +170,112 @@ export class CdcJsService implements OnDestroy {
   /**
    * Trigger CDC User registration and log in using CDC APIs.
    *
-   * @param email
-   * @param password
+   * @param user: UserSignUp
    */
-  registerUserWithoutScreenSet(user: UserSignUp) {
-    if (user.uid && user.password) {
-      (this.winRef.nativeWindow as { [key: string]: any })?.[
-        'gigya'
-      ]?.accounts?.initRegistration({
-        callback: (response: any) => {
-          this.zone.run(() => this.onInitRegistrationHandler(user, response));
-        },
-      });
-    }
+  registerUserWithoutScreenSet(user: UserSignUp): Observable<Boolean> {
+    return new Observable<Boolean>((initRegistration) => {
+      if (user.uid && user.password) {
```

Comment:
For this function to be bullet-proof, it would be good to emit an error, when this condition is negative. Otherwise the subscriber of this method will never get any value and his subscription will be pending forever

Reply:
fixed in https://github.com/SAP/spartacus/pull/16160